src/api/entities/CurrentIdentity.ts

```diff
@@ -82,11 +86,20 @@ export class CurrentIdentity extends Identity {
       context,
     } = this;
 
-    const [, ...numberedPortfolios] = await portfolios.getPortfolios();
+    const getPortfolios = await portfolios.getPortfolios();
+
+    const [ownedCustodiedPortfolios, custodiedPortfolios] = await Promise.all([
+      P.filter(getPortfolios, async portfolio => {
+        const { did: custodianIdentityDid } = await portfolio.getCustodian();
+        return custodianIdentityDid === did;
+      }),
+      this.portfolios.getCustodiedPortfolios(),
+    ]);
+
+    const allPortfolios = [...ownedCustodiedPortfolios, ...custodiedPortfolios];
 
     const portfolioIds: PortfolioId[] = [
-      { did },
-      ...numberedPortfolios.map(({ id: number }) => ({ did, number })),
+      ...(await P.map(allPortfolios, portfolio => portfolioLikeToPortfolioId(portfolio, context))),
     ];
```

Comment:
You can just do `const portfolioIds = await P.map(allPortfolios, portfolio =>  portfolioLikeToPortfolioId(portfolio, context))`